assets/scene_renderer/shaders/Config.hlsli

```diff
@@ -0,0 +1,149 @@
+#define FRAME_PARAMS_REGISTER               b1
+#define CAMERA_PARAMS_REGISTER              b2
+
+#define INSTANCE_PARAMS_REGISTER            t3
+#define MATERIAL_PARAMS_REGISTER            t4
+
+#define BRDF_LUT_SAMPLER_REGISTER           s124
+#define BRDF_LUT_TEXTURE_REGISTER           t125
+
+#define IBL_IRRADIANCE_SAMPLER_REGISTER     s126
+#define IBL_ENVIRONMENT_SAMPLER_REGISTER    s127
+#define IBL_IRRADIANCE_MAP_REGISTER         t128
+#define IBL_ENVIRONMENT_MAP_REGISTER        t144
+
+#define MATERIAL_SAMPLERS_REGISTER          s512
+#define MATERIAL_TEXTURES_REGISTER          t1024
+
+// -------------------------------------------------------------------------------------------------
+// Params Structs
+// -------------------------------------------------------------------------------------------------
+// size = 8
+struct FrameParams
+{
+    uint32_t frameIndex; // offset = 0
+    float    time;       // offset = 4
+};
+
+// size = 96
+struct CameraParams
+{
+    float4x4 viewProjectionMatrix; // offset = 0
+    float3   eyePosition;          // offset = 64
+    float    nearDepth;            // offset = 76
+    float3   viewDirection;        // offset = 80
+    float    farDepth;             // offset = 92
+};
+
+// size = 128
+struct InstanceParams
+{
+    float4x4 modelMatrix;        // offset = 0  (Object space to world space)
+    float4x4 inverseModelMatrix; // offset = 64 (World space to object space)
+};
+
+// -------------------------------------------------------------------------------------------------
+// Vertex Inputs 
+//
+// Uses fixed location numbers to make it easier on the client
+// side Vulkan code if there's gaps due to unused inputs.
+//
+// The #ifdef statements are required currently because Direct3D
+// doesn't seem to dead code eliminate unused inputs. So we have
+// to let the shader implementations control which attributes
+// they want to enable.
+//
+// -------------------------------------------------------------------------------------------------
+struct StandardVertexInput
+{
+    DECLARE_LOCATION(0) float3 PositionOS : POSITION;
+
+#if defined(ENABLE_VTX_ATTR_TEXCOORD)
+    DECLARE_LOCATION(1) float2 TexCoord : TEXCOORD;
+#endif
+
+#if defined(ENABLE_VTX_ATTR_NORMAL)  
+    DECLARE_LOCATION(2) float3 Normal : NORMAL;
+#endif
+
+#if defined(ENABLE_VTX_ATTR_TANGENT)
+    DECLARE_LOCATION(3) float4 Tangent : TANGENT;
+#endif
+
+#if defined(ENABLE_VTX_ATTR_COLOR)
+    DECLARE_LOCATION(4) float3 Color : COLOR;
+#endif
+};
+
+// -------------------------------------------------------------------------------------------------
+// Vertex Outputs
+//
+// Uses fixed location numbers to make it easier on the client
+// side Vulkan code if there's gaps due to unused inputs.
+//
+// The #ifdef statements are required currently because Direct3D
+// doesn't seem to dead code eliminate unused inputs. So we have
+// to let the shader implementations control which attributes
+// they want to enable.
```

Comment:
It depends :)

If both stages (producer, and consumer) don't use it, yes, we'd want it to be stripped.
But this require using SPIR-V tools manually to list the used interfaces, and do a cross-stage optimization.
In your case, each shader is looked up individually.

Meaning if the consumer don't use an input, and you strip it, there will be an interface mismatch between the producer shader and consumer shader. And this can cause the pipeline build to fail.

Reply:
updated